src/utils/iota.js

```diff
@@ -149,6 +149,21 @@ export const pollIotaProgress = (datamap, iotaProvider, progCb) =>
     }, POLL_INTERVAL);
   });
 
+export const pollMetadata = (handle, iotaProviders) => {
+  return new Promise((resolve, reject) => {
+    const poll = setIntervalAndExecute(() => {
+      getMetadata(handle, iotaProviders)
+        .then(res => {
+          clearInterval(poll);
+          resolve(res);
+        })
+        // TODO: Continue only if "File does not exist" error.
```

Comment:
This should be a priority, that was my only concern with doing it before. I dont want it to poll forever and a user to get confused